File: Java/FileSize.java
// FileSize.java

// サイズ情報用のクラス
public class FileSize {

    public FileSize(String fileName, long size) {
        this.fileName = fileName;
        this.size = size;
    }

    private String fileName;

    public String getFileName() {
        return this.fileName;
    }

    private long size;

    public long getSize() {
        return this.size;
    }

    public static FileSize parse(String line) {
        String[] items = line.split("\t");
        if (items.length < 2) {
            return null;
        }

        return new FileSize(items[0], Long.valueOf(items[1]));
    }

    @Override
    public String toString() {
        return String.format("[FileSize: FileName=%s, Size=%d]", fileName, size);
    }

}

File: C#/FileSize.cs
using System;

public class FileSize
{
	public string FileName { get; set; }

	public long Size { get; set; }
	
	public static FileSize Parse (string line)
	{
		string[] items = line.Split ('\t');
		if (items.Length < 2) {
			return null;
		}
		
		return new FileSize () { FileName = items [0], Size = long.Parse (items [1]) };
	}

	public override string ToString ()
	{
		return string.Format ("[FileSize: FileName={0}, Size={1}]", FileName, Size);
	}
}

